From cc75588a7aa5e3f580148e4339ebb1caaefea889 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Fri, 19 Mar 2021 12:00:28 +0100
Subject: [PATCH 1/7] file_uds: silence unused result warnings

We don't seem to care if these fail, so silence them in a way that works
with GCC.

Silences the following errors:

|     Compiling[ release ]: BigList::biglist_lock.c
| ../../src/onlplib/module/src/file_uds.c: In function 'eventfd_read__':
| ../../src/onlplib/module/src/file_uds.c:63:5: error: ignoring return value of 'read', declared with attribute warn_unused_result [-Werror=unused-result]
|      read(fd, &val, sizeof(val));
|      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
| ../../src/onlplib/module/src/file_uds.c: In function 'eventfd_write__':
| ../../src/onlplib/module/src/file_uds.c:57:5: error: ignoring return value of 'write', declared with attribute warn_unused_result [-Werror=unused-result]
|      write(fd, &val, sizeof(val));
|      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
| cc1: all warnings being treated as errors

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 packages/base/any/onlp/src/onlplib/module/src/file_uds.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/packages/base/any/onlp/src/onlplib/module/src/file_uds.c b/packages/base/any/onlp/src/onlplib/module/src/file_uds.c
index bd344d89..55e7a479 100644
--- a/packages/base/any/onlp/src/onlplib/module/src/file_uds.c
+++ b/packages/base/any/onlp/src/onlplib/module/src/file_uds.c
@@ -54,13 +54,13 @@ static void
 eventfd_write__(int fd)
 {
     uint64_t val = 1;
-    write(fd, &val, sizeof(val));
+    (void)!write(fd, &val, sizeof(val));
 }
 static void
 eventfd_read__(int fd)
 {
     uint64_t val;
-    read(fd, &val, sizeof(val));
+    (void)!read(fd, &val, sizeof(val));
 }
 
 
-- 
2.31.0

